### PR TITLE
Add idl_test helper to idlharness.js

### PR DESCRIPTION
**Pull request by lukebjerring**
`pointerlock` example attached to the PR demonstrates the removed boilerplate (common across almost all idlharness tests).

### CODE REVIEW

File: resources/idlharness.js [hunk 1]
```diff
@@ -3013,4 +3013,41 @@ function IdlTypedef(obj)
...
 
 }());
+
+/**
+ * idl_test is a promise_test wrapper that handles the fetching of the IDL,
+ * avoiding repetitive boilerplate.
+ *
+ * @param {String|String[]} srcs Spec name(s) for source idl files (fetched from
+ *      /interfaces/{name}.idl).
+ * @param {String|String[]} deps Spec name(s) for dependency idl files (fetched
```
Comment (kereliuk) [inline resources/idlharness.js]: Do these need to have a specific order for when they are added in the for loop below?
If so, we should document that order.

File: pointerlock/interfaces.window.js [hunk 2]
```diff
@@ -5,22 +5,14 @@
...
-  idl_array.add_objects({
-    Document: ["window.document"],
-    Element: ["window.document.documentElement"],
-    MouseEvent: ["new MouseEvent('foo')"]
-  });
-  idl_array.test();
-}, 'pointerlock interfaces.');
+idl_test(
+  ['pointerlock'],
+  ['uievents', 'html', 'dom'],
```
Comment (kereliuk) [inline pointerlock/interfaces.window.js]: The dependency idls are added in a different order here than in the previous revision. Does this make a difference?

**Review by kereliuk — COMMENTED** [review]
Made some ordering comments, otherwise looks good!

File: resources/idlharness.js (hunk 1 above)
Comment (lukebjerring) [reply resources/idlharness.js]: Yes, and done.

File: pointerlock/interfaces.window.js (hunk 2 above)
Comment (lukebjerring) [reply pointerlock/interfaces.window.js]: Actually, if you look at https://github.com/web-platform-tests/wpt/pull/11833/files/e3f9f7b704f6f50514f1db4dfaf16747cb5f7c3d#diff-0375a5322ffb1328cfc6a75c965c8ea2L16 `uievents` was first. I double-checked and all the tests are happily passing with this new order.

Comment (kereliuk) [reply pointerlock/interfaces.window.js]: uievents is still first but dom and html are in reverse order, although if all tests pass in the new order should be good

**Review by kereliuk — APPROVED** [review]

File: resources/idlharness.js [hunk 3]
```diff
@@ -3013,4 +3013,43 @@ function IdlTypedef(obj)
...
+ *      each source will only be included if they're already know to be a
+ *      dependency (i.e. have already been seen).
+ * @param {Function} setup_func Function for extra setup of the idl_array, such
+ *      as adding objects. Do not call idl_array.test() in the setup; it is
+ *      called by this function (idl_test).
+ */
+function idl_test(srcs, deps, setup_func, test_name) {
+    return promise_test(function (t) {
+        var idl_array = new IdlArray();
+        srcs = (srcs instanceof Array) ? srcs : [srcs] || [];
```
Comment (jgraham) [inline resources/idlharness.js]: `Array.isArray`?

Comment (jgraham) [reply resources/idlharness.js]: Also the bit after the `||` seems like a use case for default parameters? Unless we are keen to make this work with older ES engines, which I guess is not a bad idea.

File: resources/idlharness.js [hunk 4]
```diff
@@ -3013,4 +3013,43 @@ function IdlTypedef(obj)
...
+        deps = (deps instanceof Array) ? deps : [deps] || [];
+        return Promise.all(
+            srcs.concat(deps).map(function(i) {
```
Comment (jgraham) [inline resources/idlharness.js]: `i` is not a super-clear name

File: resources/idlharness.js [hunk 5]
```diff
@@ -3013,4 +3013,43 @@ function IdlTypedef(obj)
...
+                return fetch('/interfaces/' + i + '.idl').then(function(r) {
```
Comment (jgraham) [inline resources/idlharness.js]: Isn't the most common style to line up then conditions on `.`, like

```
foo
  .then(()=>{})
  .then(()=>{})
```

?

File: resources/idlharness.js (hunk 3 above)
Comment (lukebjerring) [reply resources/idlharness.js]: yeah, recently a PR was flagged by @Ms2ger as it being a better outcome to support older engines, so this PR is written attempting to be backward-friendly.

File: resources/idlharness.js (hunk 5 above)
Comment (lukebjerring) [reply resources/idlharness.js]: Followed up style issues in https://github.com/web-platform-tests/wpt/pull/11836

**Pull request merged**